tests/integ_tests/inference/native_inference/embedding_models/test_large_multilingual_clip_model_encode.py

```diff
@@ -0,0 +1,202 @@
+from marqo.inference.native_inference.load_model import load_model, clear_loaded_models
+
+LARGE_MULTILINGUAL_CLIP_TEST_MODELS = [
+    "multilingual-clip/XLM-Roberta-Large-Vit-L-14",
+    "multilingual-clip/XLM-Roberta-Large-Vit-B-16Plus",
+    "multilingual-clip/LABSE-Vit-L-14"
+]
+
+@pytest.mark.largemodel
+@parameterized_class([{"model_name": model_name} for model_name in LARGE_MULTILINGUAL_CLIP_TEST_MODELS])
+class TestLargeMultilingualCLIPModelEncode(InferenceTestCase):
+    """
+    Tests for multilingual CLIP models, on a CPU device.
+    """
+
+    model_name: str # A class variable to store the model name that will be populated by the parameterized decorator
+    device = "cuda"
+    embeddings_test_texts = [
+        "hello",
+        "This is a test sentence"
+    ]
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        clear_loaded_models()
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        current_file = Path(__file__).resolve()
+        target_dir = current_file.parent.parent.parent
+        json_file = target_dir / "embeddings_reference" / "embeddings_multilingual_clip_python_3_9.json"
+        if not os.path.exists(json_file):
+            raise FileNotFoundError(f"File {json_file} not found, which is needed to compare embeddings.")
+
+        with open(json_file, 'r') as f:
+            cls.multilingual_clip_embeddings_reference = json.load(f)
+
+    def setUp(self):
+        super().setUp()
+        self.model = load_model(
+            self.model_name,
+            model_properties=self.get_model_properties_from_registry(self.model_name),
+            model_auth=None,
+            device=self.device
+        )
+        self.eps = 1e-6
+
+    def test_embeddings_regression(self):
+        try:
+            self.model_embeddings_reference = self.multilingual_clip_embeddings_reference[self.model_name]
+        except KeyError:
+            raise KeyError(
+                f"Model {self.model_name} not found in the reference embeddings. "
+                f"Please run the test to generate the reference embeddings."
+            )
+
+        for text in self.embeddings_test_texts:
+            with self.subTest(f"Test text: {text}"):
+                embeddings_reference = np.array(self.model_embeddings_reference[text]).reshape(-1)
+                pipeline_embeddings = self.encode_content_helper(
+                    content=[text],
+                    model_name=self.model_name,
+                    modality=Modality.TEXT,
+                    device=self.device,
+                    normalize_embeddings=False
+                )
+
+                embeddings_difference = self.calculate_embeddings_difference(
+                    embeddings_reference, pipeline_embeddings[0]
+                )
+                self.assertTrue(embeddings_difference < 1e-4, embeddings_reference)
```

Comment:
Where did we get 1e-4?

Reply:
This is confirmed by Jesse as a general acceptable difference.